src/energyplus/Test/CoilSystemCoolingWater_GTest.cpp

```diff
@@ -0,0 +1,329 @@
+
+#include "../../model/CoilSystemCoolingWater.hpp"
+#include "../../model/CoilSystemCoolingWater_Impl.hpp"
+#include "../../model/Schedule.hpp"
+#include "../../model/Schedule_Impl.hpp"
+#include "../../model/CoilCoolingWater.hpp"
+#include "../../model/CoilCoolingWater_Impl.hpp"
+
+#include "../../model/AirLoopHVAC.hpp"
+#include "../../model/AirLoopHVACUnitarySystem.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem.hpp"
+#include "../../model/AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "../../model/ControllerOutdoorAir.hpp"
+#include "../../model/ControllerOutdoorAir_Impl.hpp"
+#include "../../model/PipeAdiabatic.hpp"
+#include "../../model/PipeAdiabatic_Impl.hpp"
+
+#include "../../utilities/idf/Workspace.hpp"
+#include "../../utilities/idf/IdfObject.hpp"
+#include "../../utilities/idf/WorkspaceObject.hpp"
+#include "../../utilities/idf/IdfExtensibleGroup.hpp"
+#include "../../utilities/idf/WorkspaceExtensibleGroup.hpp"
+
+// E+ FieldEnums
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/CoilSystem_Cooling_Water_FieldEnums.hxx>
+#include <utilities/idd/Coil_Cooling_Water_FieldEnums.hxx>
+#include <utilities/idd/AirLoopHVAC_FieldEnums.hxx>
+#include <utilities/idd/BranchList_FieldEnums.hxx>
+#include <utilities/idd/Branch_FieldEnums.hxx>
+#include <utilities/idd/AirLoopHVAC_UnitarySystem_FieldEnums.hxx>
+#include <utilities/idd/Controller_WaterCoil_FieldEnums.hxx>
+#include <utilities/idd/Controller_OutdoorAir_FieldEnums.hxx>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_CoilSystemCoolingWater_Unitary) {
+
+  ForwardTranslator ft;
+
+  Model m;
+
+  CoilSystemCoolingWater coilSystemCoolingWater(m);
+
+  coilSystemCoolingWater.setName("My CoilSystemCoolingWater");
+  Schedule availabilitySchedule = m.alwaysOnDiscreteSchedule();
+  EXPECT_TRUE(coilSystemCoolingWater.setAvailabilitySchedule(availabilitySchedule));
+  EXPECT_TRUE(coilSystemCoolingWater.setDehumidificationControlType("CoolReheat"));
+  EXPECT_TRUE(coilSystemCoolingWater.setRunonSensibleLoad(false));  // Opposite from IDD default
+  EXPECT_TRUE(coilSystemCoolingWater.setRunonLatentLoad(true));     // Opposite from IDD default
+  EXPECT_TRUE(coilSystemCoolingWater.setMinimumAirToWaterTemperatureOffset(1.0));
+  EXPECT_TRUE(coilSystemCoolingWater.setEconomizerLockout(false));  // Opposite from IDD default
+  EXPECT_TRUE(coilSystemCoolingWater.setMinimumWaterLoopTemperatureForHeatRecovery(1.2));
+  CoilCoolingWater companionCoilUsedForHeatRecovery(m);
+  EXPECT_TRUE(coilSystemCoolingWater.setCompanionCoilUsedForHeatRecovery(companionCoilUsedForHeatRecovery));
+
+  // put it inside a Unitary, and put Unitary on an AirLoopHVAC
+  AirLoopHVACUnitarySystem unitary(m);
+  EXPECT_FALSE(unitary.setCoolingCoil(coilSystemCoolingWater));
```

Comment:
Unlike CoilSystemCoolingWaterHeatExchangerAssisted, the CoilSystemCoolingWater object doesn't have `\reference CoolingCoilsWater` on its Name field.